src/dlgProfilePreferences.cpp

```diff
@@ -2093,6 +2119,78 @@ void dlgProfilePreferences::slot_logFileNameFormatChange(const int index)
     label_logFileNameExtension->setVisible(isShown);
 }
 
+void dlgProfilePreferences::slot_configDirOptionChange(const int index)
+{
+    Q_UNUSED(index);
+
+    // Show options to set the config directory if 'Select a
+    // directory...' is selected
+    bool isShown = index == 1;
+    label_setConfigDir->setVisible(isShown);
+    lineEdit_configDir->setVisible(isShown);
+    toolButton_setConfigDir->setVisible(isShown);
+
+    switch (index) {
+    case 0:
+        // Clear mConfigDirPath if 'Default' is selected
+        mConfigDirPath.clear();
+        lineEdit_configDir->setText(QDir::homePath() + "/.config/mudlet");
+        break;
+    case 1:
+        // Set mConfigDirPath to the value of lineEdit_configDir if
+        // 'Select a directory' is selected
+        mConfigDirPath = lineEdit_configDir->text();
+        break;
+    case 2:
+        // Set mConfigDirPath to a relative path if 'Relative to the
+        // application file' is selected
+        mConfigDirPath = ".";
+        lineEdit_configDir->setText(QCoreApplication::applicationDirPath() + "/config");
+        break;
+    }
+}
+
+void dlgProfilePreferences::slot_setConfigDir()
+{
+    QString startDir;
+    if (lineEdit_configDir->text().isEmpty() ||
+        lineEdit_configDir->text() == QStringLiteral("%1/.config/mudlet").arg(QDir::homePath())) {
+        // Set the directory the QFileDialog starts in to the default
+        // directory if the line edit is empty or equal to Mudlet's
+        // default configuration directory
+        startDir = QStringLiteral("%1/.config").arg(QDir::homePath());
+    } else if (lineEdit_configDir->text() ==
+               QStringLiteral("%1/config").arg(QCoreApplication::applicationDirPath())) {
+        // Use the contents of the line edit if set to the path
+        // relative to the application file
+        startDir = lineEdit_configDir->text();
+    } else {
+        // Start in the parent directory if a custom directory is set
+        QFileInfo configDirInfo(lineEdit_configDir->text());
+        QDir configDirParent(configDirInfo.dir());
+        startDir = configDirParent.absolutePath();
+    }
+
+    QString dir = QFileDialog::getExistingDirectory(this,
```

Comment:
:information_source: Please add a fourth argument including the `QFileDialog::DontUseNativeDialog` flag as otherwise Windows users will not see any files in the directory selection dialogue so won't be able to tell if they are going to clobber anything already existing in the selected location.

Reply:
I don't agree with this, let's use the non-native dialog if we get feedback from the users that they're clobbering stuff by accident